web/client/epics/search.js

```diff
@@ -335,5 +335,5 @@ export const delayedSearchEpic = (action$) =>
                 .switchMap(() => {
                     return Rx.Observable.of(searchLayerWithFilter({layer: name, "cql_filter": cqlFilter}));
                 })
-                .takeUntil(action$.ofType(LAYER_LOAD));
+                .takeUntil(action$.ofType(LAYER_LOAD).filter(({ layerId }) => layerId.startsWith(name)));
```

Comment:
Why `takeUntil(LAYER_LOAD)` ? 
It looks to me like it stops emitting when layer starts to load. I think it is a wrong check to do.
I'd use a `take(1)` instead, after filter by name. 
`takeUntil` can stop it at all and if you want to make sure that this flow is not pending on future changes, use `takeUntil` with page change.

Moreover I think this can be more an epic of the queryParams itself, so I think the action and the epic should be moved in `epics (actions) /queryparams.js`

Add a unit test to check this issue.

Reply:
Thanks @offtherailz, you're right, `takeUntil` didn't work correctly, `take(1`) works good with extended test coverage.
I've kept changes inside search epics/actions just because this feature doesn't make sense without `Search` plugin being active; another required epic simply won't be available to fulfill workflow in this case. Besides, adding it to `queryparams` epics will require epic to be added as one of the `appEpics` inside `main.jsx`; we don't have `queryparams` actions right now at all.